build_runner/lib/src/entrypoint/options.dart

```diff
@@ -422,25 +442,24 @@ class _TestCommand extends BuildRunnerCommand {
   @override
   Future<int> run() async {
     _SharedOptions options;
-    String outputDir;
+    // We always run our tests in a temp dir.
+    var tempPath = Directory.systemTemp
```

Comment:
This does make `-o` be wasted work now with the `test` command, should we warn about that if somebody passes it?